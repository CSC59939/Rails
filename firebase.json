{
  "functions": {
    "predeploy": [
      "npm --prefix \"$RESOURCE_DIR\" run lint"
    ]
  },
  "hosting": {
    "public": "website/build",
    "ignore": [
      "firebase.json",
      "**/.*",
      "**/node_modules/**"
    ]
  }
}
