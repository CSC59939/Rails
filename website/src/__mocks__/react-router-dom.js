import React from 'react';
const rrd = require('react-router-dom');
rrd.BrowserRouter = ({children}) => <div>{children}</div>
module.exports = rrd;
// this is a manual mock of Jest, which it redefine the browserRouter, since it will conflict with memory router.
