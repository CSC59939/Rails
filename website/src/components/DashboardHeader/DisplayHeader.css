.Header {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  z-index: 2;
  box-shadow: 0 0 8px rgba(100, 100, 100, 0.4);
  padding: 3rem !important;
  width: 100% !important;
  max-width: 100% !important;
}
