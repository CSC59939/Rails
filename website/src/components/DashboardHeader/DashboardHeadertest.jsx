import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {DashboardHeader} from '..';
import config from '../../views/firebaseconfig';
import firebase from 'firebase/app';

firebase.initializeApp(config);

configure({ adapter: new Adapter() });

describe("DashboardHeader", () => {
  it('Render DashboardHeader without any eror', () => { // eslint-disable-line no-undef
    expect(renderer.create(<DashboardHeader />).toJSON()).toMatchSnapshot();
  });
})
