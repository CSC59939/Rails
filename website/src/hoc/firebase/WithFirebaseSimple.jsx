import React, { PureComponent } from 'react';
import firebase from 'firebase/app';
import 'firebase/auth';

function WithFirebaseSimple(App) {
  return class extends PureComponent {
    render() {
      return (
        <App
          {...this.props}
          firebase={firebase}
        />
      );
    }
  };
}

export default WithFirebaseSimple;
