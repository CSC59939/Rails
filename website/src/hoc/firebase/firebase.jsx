import React, { PureComponent } from 'react';
import firebase from 'firebase/app';
import 'firebase/auth';

function withFirebase(App) {
  return class extends PureComponent {
    constructor(props) {
      super(props);
      this.state = {
        signedin: false,
      };
      this.firebaseListener = null;
      this.authStateListener = this.authStateListener.bind(this);
      this.signoutHandler = this.signoutHandler.bind(this);
      if (firebase.app()) this.authStateListener();
    }

    componentDidMount() {
      // if (firebase.app()) this.authStateListener();
    }

    componentWillUnmount() {
      if (this.firebaseListener) {
        this.firebaseListener();
      }
      this.authStateListener = undefined;
    }

    authStateListener() {
      this.firebaseListener = firebase.auth().onAuthStateChanged((user) => {
        if (user) {
          this.setState({ signedin: true });
        } else {
          this.setState({ signedin: false });
        }
      });
    }

    signoutHandler() {
      console.log('Signing out');
      firebase.auth().signOut();
      this.setState({ signedin: false });
    }

    render() {
      const {
        signedin,
      } = this.state;
      return (
        <App
          {...this.props}
          signoutHandler={this.signoutHandler}
          signedin={signedin}
        />
      );
    }
  };
}

export default withFirebase;
