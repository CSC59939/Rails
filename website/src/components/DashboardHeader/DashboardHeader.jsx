import React, { PureComponent } from 'react';
import {
  Layout,
} from 'antd';
import PropTypes from 'prop-types';
import firebase from 'firebase/app';
import { ProfileInfo, HeaderIcons } from '..';
import './DisplayHeader.css';

const { Header } = Layout;

class DashboardHeader extends PureComponent {
  static propTypes = {
    showDrawer: PropTypes.func,
    teacher: PropTypes.bool,
  }

  static defaultProps = {
    showDrawer: () => console.log('Never passed showDrawer'),
    teacher: false,
  }

  render() {
    const {
      showDrawer,
      teacher,
    } = this.props;

    const TeacherIcons = [
      {
        type: 'add_circle',
        onClick: () => console.log('add event'),
        key: 'add_circle',
      },
      {
        type: 'add_alert',
        onClick: () => console.log('add notification'),
        key: 'add_alert',
      },
    ];

    const Icons = [
      {
        type: 'notifications_active',
        onClick: showDrawer,
        key: 'notifications_active',
      },
      {
        type: 'settings',
        onClick: () => { window.location = '/dashboard/profile'; },
        key: 'settings',
      },
    ];
    if (teacher) {
      Icons.unshift(...TeacherIcons);
    }
    return (
      <Header className="Header">
        <ProfileInfo
          name={firebase.auth().currentUser ? firebase.auth().currentUser.displayName : 'User Name'}
          email={firebase.auth().currentUser ? firebase.auth().currentUser.email : 'email@domain.com'}
        />
        <HeaderIcons Icons={Icons} />
      </Header>
    );
  }
}

export default DashboardHeader;
