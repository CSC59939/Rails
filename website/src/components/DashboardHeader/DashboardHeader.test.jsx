import React from 'react';
import renderer from 'react-test-renderer';
import {
  configure, shallow,
} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { MemoryRouter } from 'react-router-dom';
import firebase from 'firebase';
import DashboardHeader from './DashboardHeader';

configure({ adapter: new Adapter() });

describe('DashboardHeader', () => {
  it('Expect DashboardHeader to match snapshot', () => {
    jest.spyOn(firebase, 'auth')
      .mockImplementation(() => ({
        currentUser: {
          displayName: 'testDisplayName',
          email: 'test@test.com',
        },
      }));
    const component = renderer.create(
      <MemoryRouter>
        <DashboardHeader showDrawer={() => {}} teacher />
      </MemoryRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
