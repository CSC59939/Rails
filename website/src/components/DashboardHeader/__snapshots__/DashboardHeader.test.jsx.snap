// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardHeader Expect DashboardHeader to match snapshot 1`] = `
<div
  className="Header ant-layout-header"
>
  <div
    className="ProfileInfo"
  >
    <a
      href="/dashboard"
    >
      <img
        alt="Rails"
        className="header-logo"
        src="logo.svg"
      />
    </a>
    <div
      className="rightSide"
    >
      <p
        className="name"
      >
        testDisplayName
      </p>
      <p
        className="email"
      >
        test@test.com
      </p>
    </div>
  </div>
  <div
    className="HeaderIcons"
  >
    <button
      className="IconButton"
      onClick={[Function]}
      type="button"
    >
      <i
        className="material-icons"
        style={null}
      >
        add_circle
      </i>
    </button>
    <button
      className="IconButton"
      onClick={[Function]}
      type="button"
    >
      <i
        className="material-icons"
        style={null}
      >
        settings
      </i>
    </button>
    <a
      href="/signout"
      onClick={[Function]}
    >
      <button
        className="IconButton"
        onClick={[Function]}
        type="button"
      >
        <i
          className="material-icons"
          style={null}
        >
          exit_to_app
        </i>
      </button>
    </a>
  </div>
</div>
`;
